### PR TITLE
Use immutable sets in YoutubeParsingHelper.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
----
* Bump the required Java version to 11.
* Use the immutable `Set` factory methods in `YoutubeParsingHelper`.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,8 +5,8 @@ allprojects {
     compileJava.options.encoding = 'UTF-8'
     compileTestJava.options.encoding = 'UTF-8'
 
-    sourceCompatibility = 1.8
-    targetCompatibility = 1.8
+    sourceCompatibility = 9
+    targetCompatibility = 9
```
Comment (AudricV) [inline build.gradle:9]: You should have precised that your changes require Java 9!

Comment (opusforlife2) [reply build.gradle:9]: ~~precised~~ specified

Comment (Isira-Seneviratne) [reply build.gradle:9]: Sorry about that. I updated the description.

Comment (Isira-Seneviratne) [reply build.gradle:9]: I didn't use Java 11 methods in this PR, but I guess Java 11 could be used instead.

**Review by FireMasterK — APPROVED** [review]
Looks mostly good, any particular reason why we're using `Locale.ROOT` and not `Locale.US` or `Locale.ENGLISH`? Since these are URLs, I don't think it matters much, does it?

**Comment by Isira-Seneviratne** [comment]
> Looks mostly good, any particular reason why we're using `Locale.ROOT` and not `Locale.US` or `Locale.ENGLISH`? Since these are URLs, I don't think it matters much, does it? 

I used the suggestions provided in Android Studio.

**Pull request merged**